apm-agent-plugins/apm-scheduled-annotation-plugin/src/test/java/co/elastic/apm/agent/spring/scheduled/TimerTaskInstrumentationTest.java

```diff
@@ -56,14 +55,13 @@ void testTimerTask_scheduleWithFixedRate() throws InterruptedException {
 
     @Test
     void testTimerTask_scheduleWithFixedDelay() throws InterruptedException {
-        reporter.reset();
         TestTimerTask timerTask = new TestTimerTask();
         Timer timer = new Timer("Timer");
         timer.schedule(timerTask, 1L, 10L);
 
         reporter.awaitUntilAsserted(1000L, () -> {
-            timer.cancel();
             assertThat(reporter.getTransactions()).isNotEmpty();
+            timer.cancel();
```

Comment:
Test is failing. Maybe revert to happen before the assertion and wait 20ms before asserting.

Reply:
I rewrote the test a little. It should be much more deterministic and resilient now.